pymc3/distributions/continuous.py

```diff
@@ -2224,6 +2262,24 @@ def logcdf(value, alpha, beta):
         )
 
 
+class PyMC3HalfCauchyRV(RandomVariable):
+    name = "halfcauchy"
+    ndim_supp = 0
+    ndims_params = [0]
+    dtype = "floatX"
+    _print_name = ("C**+", "\\operatorname{C^{+}}")
+
+    def __call__(self, beta=1.0, size=None, **kwargs):
+        return super().__call__(beta, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(cls, rng, beta, size):
+        return stats.halfcauchy.rvs(scale=beta, random_state=rng, size=size)
+
+
+halfcauchy = PyMC3HalfCauchyRV()
```

Comment:
Again, creating a new class for a _reduced_ parameter space doesn't make sense unless it's absolutely necessary.

Here, it looks like a default `loc` of zero will work, and not require a new class.

Reply:
The problem is that the original class expects two parameters (or at least that's what I figured out from trying to just inherit)

Reply:
Then we make the `Distribution.dist` interface take only the used parameter, `scale`/`beta`, and set the unused `loc` parameter to zero.

Reply:
Don't we still need to at least invert the order of the arguments in `rng_fn`. Sounds like a messier solution, but I'm happy to try

Original Aesara HalfCauchyRV
```python
class HalfCauchyRV(RandomVariable):
    name = "cauchy"
    ndim_supp = 0
    ndims_params = [0, 0]
    dtype = "floatX"
    _print_name = ("C**+", "\\operatorname{C^{+}}")

    def __call__(self, loc=0.0, scale=1.0, size=None, **kwargs):
        return super().__call__(loc, scale, size=size, **kwargs)

    @classmethod
    def rng_fn(cls, rng, loc, scale, size):
        return stats.halfcauchy.rvs(loc=loc, scale=scale, random_state=rng, size=size)
```

Reply:
Remember, users are **not** creating `RandomVariable`s directly; they're using `Distribution.dist`, so that's where we need to focus.

Reply:
I think I just need to se an example where the loc parameter is ignored, set to zero by default and the relevant (on the pymc3 side) scale argument is used properly. Before this refactoring it was mistaken as the loc and the scale defaulted to 1 (that's why it failed in the reintroduced test_distributions_random.py test)

Reply:
Yeah, it sounds like there was a big underlying that issue.  I'll try to push the changes I'm alluding to in a minute.